src/components/studies/app-design/AppDesign.tsx

```diff
@@ -316,6 +332,88 @@ const AppDesign: React.FunctionComponent<
     isIrbEmailValid: true,
   })
 
+  const [errorState, setErrorState] = React.useState<ErrorStateType>({})
+
+  const handleUpdate = (updatedStudy: Study) => {
+    const formattedStudy = formatStudy(updatedStudy)
+    if (formattedStudy.contacts) {
+      for (const contact of formattedStudy.contacts) {
+        if (contact.name === '') {
+          contact.name = DEFAULT_CONTACT_NAME
+        }
+      }
+    }
+    onUpdate(formattedStudy)
+  }
+
+  const checkPhoneError = (contactLead?: Contact, irbRecord?: Contact) => {
+    const generalContactPhoneError =
+      !contactLead?.phone?.number || isInvalidPhone(generalContactPhoneNumber)
+    const irbRecordHasError =
+      !irbRecord?.phone?.number || isInvalidPhone(irbPhoneNumber)
+    setPhoneNumberErrorState({
+      isGeneralContactPhoneNumberValid: !generalContactPhoneError,
+      isIrbPhoneNumberValid: !irbRecordHasError,
+    })
+  }
+
+  const checkEmailError = (contactLead?: Contact, irbRecord?: Contact) => {
+    const generalContactEmailHasError =
+      !contactLead?.email || !isValidEmail(contactLead.email)
+    const irbRecordEmailHasError =
+      !irbRecord?.email || !isValidEmail(irbRecord.email)
+    setEmailErrorState({
+      isGeneralContactEmailValid: !generalContactEmailHasError,
+      isIrbEmailValid: !irbRecordEmailHasError,
+    })
+  }
+
+  const isContactValid = (
+    contact: Contact | undefined,
+    property: keyof Contact
+  ) => {
+    return (
+      contact && contact[property] && contact[property] !== DEFAULT_CONTACT_NAME
+    )
+  }
+
+  useEffect(() => {
+    if (!showError) return
+    const updatedErrorState = {} as ErrorStateType
+    const contactLead = study.contacts?.find(el => el.role === 'study_support')
+    const principalInvestigator = study.contacts?.find(
+      c => c.role === 'principal_investigator'
+    )
+    const irbRecord = study.contacts?.find(el => el.role === 'irb')
+    if (!isContactValid(principalInvestigator, 'name')) {
```

Comment:
here and below how about 
` updatedErrorState.leadPINameHasError = !isContactValid(principalInvestigator, 'name') ` etc the you don't need the conditions